### ISSUE TITLE
net/http: http2 client return EOF

### ISSUE THREAD

**Issue by mattn** [golang/go#13598]
I'm trying to connect to https://dav.box.com/ with golang. It seems that box.com provide http2 protocol. But go's http2 client retrun EOF always. I debugged little and I make sure this occur on h2 protocol in TLSNextProto. So currently I'm disabling h2 like below.

```go
http.DefaultTransport.(*http.Transport).TLSNextProto = make(map[string]func(string, *tls.Conn) http.RoundTripper)
```

**Comment by bradfitz** [comment]
Can you show sample code to reproduce?

Or do you literally mean just:

```go
     res, err := http.Get("https://dav.box.com/")
```

?

**Comment by bradfitz** [comment]
Indeed! Thanks for the bug report.

```
brad5k:~ $ cat get.go
```
```go
package main

import (
	"io"
	"log"
	"net/http"
	"os"
)

func main() {
	res, err := http.Get("https://dav.box.com/")
	if err != nil {
		log.Fatal(err)
	}
	log.Printf("Res: %#v", res)
	_, err = io.Copy(os.Stdout, res.Body)
	if err != nil {
		log.Fatal(err)
	}
}
```

With HTTP/2 enabled by default: (mix of error messages)
```
brad5k:~ $ go run get.go 
2015/12/13 11:19:14 Get https://dav.box.com/: EOF
exit status 1
brad5k:~ $ go run get.go 
2015/12/13 11:19:15 Get https://dav.box.com/: EOF
exit status 1
brad5k:~ $ go run get.go 
2015/12/13 11:19:17 Get https://dav.box.com/: read tcp 10.0.0.55:55327->74.112.184.85:443: read: connection reset by peer
exit status 1
```

With HTTP/2 explicitly disabled:
```
brad5k:~ $ GODEBUG=h2client=0 go run get.go
2015/12/13 11:18:13 Res: &http.Response{Status:"401 Unauthorized", StatusCode:401, Proto:"HTTP/1.1", ProtoMajor:1, ProtoMinor:1, Header:http.Header{"Content-Length":[]string{"242"}, "Date":[]string{"Sun, 13 Dec 2015 19:18:13 GMT"}, "Age":[]string{"0"}, "Connection":[]string{"keep-alive"}, "Server":[]string{"ATS"}, "Content-Type":[]string{"application/xml; charset=utf-8"}, "Www-Authenticate":[]string{"Basic realm=\"dav.box.com\""}}, Body:(*http.bodyEOFSignal)(0xc8201380c0), ContentLength:242, TransferEncoding:[]string(nil), Close:false, Trailer:http.Header(nil), Request:(*http.Request)(0xc8200ce000), TLS:(*tls.ConnectionState)(0xc82035f080)}
<?xml version="1.0" encoding="utf-8"?>
<d:error xmlns:d="DAV:" xmlns:s="http://sabredav.org/ns">
  <s:exception>Sabre_DAV_Exception_NotAuthenticated</s:exception>
  <s:message>No basic authentication headers were found</s:message>
</d:error>
```

**Comment by mattn** [comment]
Yes, I got same result.

**Comment by bradfitz** [comment]
I found the problem. https://dav.box.com does **not** speak HTTP/2. But we advertise in our TLS connection that we speak "h2" but we never mention that we also speak "http/1.1", and Box's frontend recognizes ALPN or NPN enough to know that they can't handle "h2", so they hang up on us.

The fix is to declare that we also speak "http/1.1" as a fallback.

Easy fix.

**Comment by mattn** [comment]
thank you
